Assets/Scripts/EnemyBehaviour.cs

```diff
@@ -0,0 +1,197 @@
+/// \file Framework classes for Enemy actions and implementations thereof
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Enemy's level of involvement with the player
+/// </summary>
+public enum ENEMY_STATE
+{
+    /// <summary>
+    /// Enemy is unaware of player
+    /// </summary>
+    idle = 0,
+    /// <summary>
+    /// Enemy is aware of player but not engaged in conflict
+    /// </summary>
+    aware = 1,
+    /// <summary>
+    /// Enemy is engaged in conflict with player
+    /// </summary>
+    engaged = 2
+}
+
+/// <summary>
+/// Exception thrown in the event that an ENEMY_STATE value is incompatible with a function
+/// </summary>
+public class InvalidEnemyStateException : System.ArgumentException
+{
+    /// <summary>
+    /// Throws an exception with a given error message.
+    /// </summary>
+    /// <param name="message"> Error message </param>
+    public InvalidEnemyStateException(string message) : base(message)
+    {
+    }
+}
+
+/// <summary>
+/// Abstract class providing framework for Enemy actions
+/// </summary>
+public abstract class EnemyBehaviour
+{
+    /// <summary>
+    /// Array storing information about intended use of the EnemyBehaviour in different ENEMY_STATEs
+    /// </summary>
+    /// <remarks>
+    /// Given an ENEMY_STATE state, has the property that _use_case[(int)state] iff the EnemyBehaviour is designed for use with state.
+    /// An EnemyBehaviour is not designed for use with any state by default, so _use_case is initialized as all false.
+    /// </remarks>
+    protected bool[] _use_case = new bool[3];
+
+    /// <summary>
+    /// Enemy's Rigidbody2D
+    /// </summary>
+    protected Rigidbody2D _enemyBody;
+
+    /// <summary>
+    /// Checks if an ENEMY_STATE is an intended use case for the EnemyBehaviour
+    /// </summary>
+    /// <param name="state"> The ENEMY_STATE in which the EnemyBehaviour will be used by an Enemy. Can be set to null to skip check. </param>
+    /// <exception cref="InvalidEnemyStateException"> Thrown if state is not an intended use case for the EnemyBehaviour </exception>
+    protected void _CheckUseCase(ENEMY_STATE? state)
+    {
+        if (state is ENEMY_STATE state_val
+        && !this._use_case[(int)state_val])
```

Comment:
```suggestion
        if (
            state is ENEMY_STATE state_val
            && !this._use_case[(int)state_val]
         )
```

Reply:
Trying to use parentheses consistently like curly braces?

If so, did you mean to indent the closing parenthesis an extra space? it's in the column with the "f" of "if" instead of the "i".

Reply:
Did not mean the extra space